src/components/tools/text-tool.tsx

```diff
@@ -69,11 +58,39 @@ export default class TextToolComponent extends BaseComponent<IProps, IState> {
     if (content.type === "Text") {
       if (!readOnly) {
         content.setSlate(change.value);
+
+        // The state will have updated in response to the content change, but the content loses the cursor position
+        // So, we update here using the raw change, which still contains the cursor position
+        this.setState({ value: change.value });
       }
-      this.setState({ value: change.value });
     }
   }
 
+  public componentDidMount() {
+    const inititialTextContent = this.props.model.content as TextContentModelType;
+    this.prevText = inititialTextContent.text;
+    const initialValue = inititialTextContent.convertSlate();
+    this.setState({
+      value: initialValue
+    });
+
+    this.disposers = [];
+    this.disposers.push(autorun(() => {
+      const textContent = this.props.model.content as TextContentModelType;
+      if (this.prevText !== textContent.text) {
+        const newValue = textContent.convertSlate(this.state.value);
```

Comment:
There's something I'm not understanding here. The autorun is triggered when the text content changes. It seems odd that our response is to ignore the actual content change and instead to call `convertSlate(this.state.value)`, which seems like it would just set the new state to the previous state.